rust/feature-flags/src/flags/flag_request.rs

```diff
@@ -96,6 +100,20 @@ impl FlagRequest {
         }
     }
 
+    /// Sanitizes JSON by replacing unquoted NaN/Infinity with null, preserving quoted strings
+    /// This matches Python decide endpoint behavior: parse_constant=lambda x: None
+    fn sanitize_unquoted_non_finite_values(json_str: &str) -> String {
+        static NEG_INF_RE: Lazy<Regex> =
+            Lazy::new(|| Regex::new(r"(?i)-infinity(\s*[,}\]])").unwrap());
+        static INF_RE: Lazy<Regex> =
+            Lazy::new(|| Regex::new(r"(?i)\binfinity(\s*[,}\]])").unwrap());
+        static NAN_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r"(?i)\bnan(\s*[,}\]])").unwrap());
```

Comment:
I'm not a fan of using regex for this because it's not JSON aware. For example, what would happen with:

```json
let json_with_substrings = r#"{"api_key": "test", "distinct_id": "user", "person_properties": {"payload": "{'foo': NaN}", "message": "Infinity is a concept"}}"#;
```

Reply:
Or `{"api_key": "test", "payload": "This NaN, has a comma after it"}`

Reply:
hmm good point, I'll try something else instead of using a regex